From 7773b33b5dce073efc1ba0b26494232f40992018 Mon Sep 17 00:00:00 2001
From: Yuxi <b36102@freescale.com>
Date: Thu, 3 Nov 2011 10:25:54 +0800
Subject: [PATCH 0537/2463] ENGR00161300 MXC V4L2 capture: fix build warning

fix build warning:
> drivers/media/video/mxc/capture/mxc_v4l2_capture.c:2457: warning:
> comparison between pointer and integer

Signed-off-by: Yuxi Sun <b36102@freescale.com>
---
 drivers/media/video/mxc/capture/mxc_v4l2_capture.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/media/video/mxc/capture/mxc_v4l2_capture.c b/drivers/media/video/mxc/capture/mxc_v4l2_capture.c
index 90066d5..229e3f6 100644
--- a/drivers/media/video/mxc/capture/mxc_v4l2_capture.c
+++ b/drivers/media/video/mxc/capture/mxc_v4l2_capture.c
@@ -2454,7 +2454,7 @@ static void init_camera_struct(cam_data *cam, struct platform_device *pdev)
 	cam->ipu = ipu_get_soc(0);
 	if (cam->ipu == NULL)
 		pr_err("ERROR: v4l2 capture: failed to get ipu\n");
-	else if (cam->ipu == -ENODEV)
+	else if (cam->ipu == ERR_PTR(-ENODEV))
 		pr_err("ERROR: v4l2 capture: get invalid ipu\n");
 
 	init_MUTEX(&cam->param_lock);
-- 
1.7.7.4

